kms/kms_test.go

```diff
@@ -484,6 +484,68 @@ func TestIAMRemoveMember(t *testing.T) {
 	}
 }
 
+func TestImportEndToEnd(t *testing.T) {
+	testutil.SystemTest(t)
+
+	cryptoKeyID := fixture.RandomID()
+	t.Run("createKeyForImport", func(t *testing.T) {
+		var b bytes.Buffer
+		if err := createKeyForImport(&b, fixture.KeyRingName, cryptoKeyID); err != nil {
+			t.Fatal(err)
+		}
+
+		if got, want := b.String(), "Created key"; !strings.Contains(got, want) {
+			t.Errorf("createKeyForImport: expected %q to contain %q", got, want)
+		}
+	})
+	cryptoKeyName := fmt.Sprintf("%s/cryptoKeys/%s", fixture.KeyRingName, cryptoKeyID)
+
+	importJobID := fixture.RandomID()
+	t.Run("createImportJob", func(t *testing.T) {
+		var b bytes.Buffer
+		if err := createImportJob(&b, fixture.KeyRingName, importJobID); err != nil {
+			t.Fatal(err)
+		}
+
+		if got, want := b.String(), "Created import job"; !strings.Contains(got, want) {
+			t.Errorf("createImportJob: expected %q to contain %q", got, want)
+		}
+	})
+	importJobName := fmt.Sprintf("%s/importJobs/%s", fixture.KeyRingName, importJobID)
+
+	t.Run("checkStateImportJob", func(t *testing.T) {
```

Comment:
Why is this being run in a separate goroutine?

Reply:
If so, then please create a separate test case for each sample.

Reply:
IMO using subtests is generally a fine pattern and not overly complex. However, the typical use for them is table-driven tests where cases are independent of one another. In this case, it seems like you are maintaining state between test cases, for example if `createKeyForImport` fails then I think later cases will fail/panic also because they don't have the necessary state, which then creates more noise to debug the output.

I think what would be less confusing would be either:
1. (better) Don't use subtests; write one test and use `t.Fatalf` to fail fast if a sample that creates resources fails.
2. (best) Separate out into separate tests which do their own setup/cleanup calling library code directly, rather than relying on setup/cleanup provided by other samples.

LMK if you have more questions about this.

Reply:
The samples executed in the (single) test are all snippets that are part of a complex multi-step process.  See https://cloud.google.com/kms/docs/importing-a-key which is already done in Python.  These are really only separate samples because we want to insert prose in between them for context in the doc page.

Splitting into a single test case per sample would just end up with a ton of duplicate code, because a test for sample n necessarily needs the results of samples[1:n-1].

Reply:
In any case, I removed the subtests and used approach 1 suggested by @tritone